stl/inc/flat_map

```diff
@@ -833,19 +842,25 @@ private:
             && !_Key_compare(_STD forward<_K2>(_Y), _STD forward<_K1>(_X));
     }
 
+    auto _View_to_sort() {
+        using _Sorting_iterator = _Paring_iterator_provider<key_container_type, mapped_container_type,
+            _Paring_iterator_kind::_Sort>::_Iterator;
+        return _RANGES subrange<_Sorting_iterator>{_Sorting_iterator{_Data.keys.begin(), _Data.values.begin()},
+            _Sorting_iterator{_Data.keys.end(), _Data.values.end()}};
+    }
+
     void _Sort() {
         _Clear_flat_map_scope_guard _Guard{this};
-        auto _Zip_view = _RANGES views::zip(_Data.keys, _Data.values);
-        _RANGES sort(_Zip_view, value_compare(_Key_compare));
+        _RANGES sort(_View_to_sort(), value_compare(_Key_compare));
         _Guard._Clearable = nullptr;
     }
 
     void _Dedup() {
         _Clear_flat_map_scope_guard _Guard{this};
-        auto _Zip_view = _RANGES views::zip(_Data.keys, _Data.values);
-        auto _Subrange = _RANGES unique(
-            _Zip_view, [this](const_reference _X, const_reference _Y) { return this->_Key_equal(_X.first, _Y.first); });
-        auto _Remaining_count = _STD distance(_Zip_view.begin(), _Subrange.begin());
+        auto _Sorted_view     = _View_to_sort();
+        auto _Subrange        = _RANGES unique(_Sorted_view,
+                   [this](const_reference _X, const_reference _Y) { return this->_Key_equal(_X.first, _Y.first); });

```

Comment:
Pre-existing: The macro-like names `_X`, `_Y`, `_K1`, `_K2`, etc. will need to be changed at some point.